extension/src/popup/helpers/balance.ts

```diff
@@ -36,12 +36,20 @@ export const findAssetBalance = (
       "token" in balance && "issuer" in balance.token
         ? balance.token.issuer.key
         : "";
-    return balanceIssuer === asset.issuer;
+    const balanceCode =
+      "token" in balance && "code" in balance.token ? balance.token.code : "";
+    return balanceIssuer === asset.issuer && balanceCode === asset.code;
   }) as Exclude<AssetType, LiquidityPoolShareAsset> | undefined;
 };
 
-export const getBalanceByIssuer = (issuer: string, balances: AssetType[]) =>
-  balances.find((balance) => {
+export const getBalanceByAsset = (
```

Comment:
this fixes an issue where there could be a collision if 2 assets had the same issuer